stackslib/src/net/api/blocksimulate.rs

```diff
@@ -0,0 +1,294 @@
+
+        let block_id = StacksBlockId::from_hex(block_id_str).map_err(|_| {
+            Error::DecodeError("Invalid path: unparseable consensus hash".to_string())
+        })?;
+        self.block_id = Some(block_id);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCNakamotoBlockSimulateRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.block_id = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let block_id = self
+            .block_id
+            .take()
+            .ok_or(NetError::SendError("Missing `block_id`".into()))?;
+
+        let simulated_block_res =
+            node.with_node_state(|_network, sortdb, chainstate, _mempool, _rpc_args| {
+                let Some((tenure_id, parent_block_id)) = chainstate
+                    .nakamoto_blocks_db()
+                    .get_tenure_and_parent_block_id(&block_id)?
+                else {
+                    return Err(ChainError::NoSuchBlockError);
+                };
+
+                let staging_db_path = chainstate.get_nakamoto_staging_blocks_path()?;
+                let db_conn =
+                    StacksChainState::open_nakamoto_staging_blocks(&staging_db_path, false)?;
+                let rowid = db_conn
+                    .conn()
+                    .get_nakamoto_block_rowid(&block_id)?
+                    .ok_or(ChainError::NoSuchBlockError)?;
+
+                let mut blob_fd = db_conn
+                    .open_nakamoto_block(rowid, false)
+                    .map_err(|e| {
+                        let msg = format!("Failed to open Nakamoto block {}: {:?}", &block_id, &e);
+                        warn!("{}", &msg);
+                        msg
+                    })
+                    .unwrap();
+
+                let block = NakamotoBlock::consensus_deserialize(&mut blob_fd)
+                    .map_err(|e| {
+                        let msg = format!("Failed to read Nakamoto block {}: {:?}", &block_id, &e);
+                        warn!("{}", &msg);
+                        msg
+                    })
+                    .unwrap();
+
+                let burn_dbconn = match sortdb.index_handle_at_block(chainstate, &parent_block_id) {
+                    Ok(burn_dbconn) => burn_dbconn,
+                    Err(_) => return Err(ChainError::NoSuchBlockError),
+                };
+
+                let (block_fees, txs_receipts) = chainstate
+                    .with_simulated_clarity_tx(
+                        &burn_dbconn,
+                        &parent_block_id,
+                        &block_id,
+                        |clarity_tx| {
+                            let (block_fees, txs_receipts) =
```

Comment:
There's _a lot more_ to processing a block than processing its transactions.  Specifically, there's stuff in the block header that a transaction can access, so you'll need to do the relevant block-processing as well.  Please see `crate::net::api::postblock_proposal::NakamotoBlockProposal` for inspiration -- in particular, it uses a `NakamotoBlockBuilder` to set up the `ClarityTx` before processing transactions (and this code path uses the same code path that the consensus-critical block-processing logic uses).

Reply:
Yep, I can definitely reuse the BlockBuilder logic, but again, it interally calls inner_setup_block that calls begin_block that checks for the new block presence in the MARF